Exercise_3/FileManagerActivity.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Util;
+using Android.Views;
+using Android.Widget;
+using Exercise_3.Adapters;
+using File = Java.IO.File;
+
+namespace Exercise_3
+{
+    [Activity(Label = "FileManagerActivity", MainLauncher = true)]
+    public class FileManagerActivity : ListActivity
+    {
+        private readonly List<string> videoExtensions = new List<string> {".mp3", ".mp4", ".flv", ".3gp"};
+        private readonly List<string> imageExtensions = new List<string> {".jpg", ".png"};
+
+        private string path = File.Separator;
+
+        private string root;
+
+        private static FileAdapter adapter;
+
+        private static readonly List<string> files = new List<string>();
+        
+        protected override void OnCreate(Bundle savedInstanceState)
```

Comment:
It's minor but you can even make this class better with some changes

1. Don't make the adapter is static

```C#
private readonly List<string> files = new List<string>();
private readonly FileAdapter adapter = new FileAdapter(files);
```

2. Remove
```C#
private string root;
```
```C#
private void UpdateListView(string filePath){//Remove ListView listview because you can access it directly
   path = filePath;
  ....
}

On create 
UpdateListView(ListView, File.Separator);

On OnBackPressed()
UpdateListView(Path.GetPathRoot(path));
```

Reply:
Done